pkg/controllers/cluster/taint_manager.go

```diff
@@ -170,6 +171,10 @@ func (tc *NoExecuteTaintManager) syncBindingEviction(key util.QueueKey) error {
 	// Case 2: Need eviction after toleration time. If time is up, do eviction right now.
 	// Case 3: Tolerate forever, we do nothing.
 	if needEviction || tolerationTime == 0 {
+		err := controllerUtils.UpdateFailoverStatus(tc.Client, binding, workv1alpha2.EvictionReasonTaintUntolerated)
+		if err != nil {
+			klog.Errorf("Failed to update status with failover information. Error: %v", err)
+		}
```

Comment:
Here adds a duplicated history item, doesn't it? We recorded the history at the application controller when triggering the failover.

Reply:
I can double check, but I don’t believe so. My understanding was that this is used for cluster failover, which is separate from the application failover feature.

Reply:
> My understanding was that this is used for cluster failover, which is separate from the application failover feature.

Oh, yes, you are right.  My bad.